cli/src/args.rs

```diff
@@ -83,6 +83,9 @@ pub struct Args {
     /// Increase verbosity (can be supplied multiple times).
     #[arg(short = 'v', long = "verbose", global = true, action = ArgAction::Count)]
     pub verbosity: u8,
+    /// User kernel version to check lints against
+    #[arg(short = 'k', long = "kernel_version")]
+    pub kernel_version: Option<String>,
```

Comment:
We should represent this member as `Option<Version>` to have fully parsed arguments ready for use and not require a second parsing pass. Please take a look at the `value_parser` logic that we use for `srcs`. I think it should be possible to come up with similar logic for parsing the version.